File: reactapp-architecture/ios/RNLImageViewManager.m
//
//  RNSDImageManager.m
//  ShowSourcing
//
//  Created by linh on 4/4/19.
//  Copyright © 2019 Facebook. All rights reserved.
//

#import "React/RCTViewManager.h"

@interface RCT_EXTERN_MODULE(RNLImageViewManager, RCTViewManager)

RCT_EXPORT_VIEW_PROPERTY(source, NSDictionary)
RCT_EXPORT_VIEW_PROPERTY(size, NSDictionary)
RCT_EXPORT_VIEW_PROPERTY(scaleMode, NSString)

RCT_EXPORT_VIEW_PROPERTY(onResourceReady, RCTDirectEventBlock)
RCT_EXPORT_VIEW_PROPERTY(onLoadFailed, RCTDirectEventBlock)

@end

File: reactapp-architecture/ios/RNLImageViewManager.swift
//
//  RNSDImageManager.swift
//  ShowSourcing
//
//  Created by linh on 4/4/19.
//  Copyright © 2019 Facebook. All rights reserved.
//

import Foundation

@objc(RNLImageViewManager)
class RNLImageViewManager: RCTViewManager {
  override static func requiresMainQueueSetup() -> Bool {
    return true
  }
  
  override func view() -> UIView! {
    return RNLImageView(frame: CGRect.zero)
  }
}
